tests/test_engine.py

```diff
@@ -152,6 +172,196 @@ def test_links():
     fe.assert_called_once_with([q.Field('e')], [2])
 
 
+def test_links_requires_list():
+    db = {
+        'song': {
+           100: {'name': 'fuel',
+                 'artist_id': 1,
+                 'album_id': 10}
+        },
+        'artist': {
+            1: {'name': 'Metallica'},
+        },
+        'album': {
+            10: {'name': 'Reload'}
+        }
+    }
+
+    link_song = Mock(return_value=100)
+
+    def link_song_info(reqs: List[Tuple]):
+        return reqs
+
+    def song_fields(fields, song_ids):
+        def get_fields(song_id):
+            for f in fields:
+                yield db['song'][song_id][f.name]
+
+        return [list(get_fields(song_id)) for song_id in song_ids]
+
+    def song_info_fields(fields, ids):
+        def get_fields(id_):
+            [album_id, artist_id] = id_
```

Comment:
Alternative implementation using dict would look like:

```python
album_id = id_['album_id']
artist_id = id_['artist_id']
```
Pros:
 - this code will not broke if order of fields in `require=[...]` wlll change.
Cons:
  - ?